releng/jenkins/check-versions/Jenkinsfile

```diff
@@ -215,25 +224,30 @@ def getXtextGradlePluginVersion (branch) {
  * @param repository (Optional) Xtext repository name on GH
  */
 def getVersionFromGradleVersions (id,branch='master',repository='xtext-lib') {
-  return sh (script: "curl -sS https://raw.githubusercontent.com/eclipse/${repository}/${branch}/gradle/versions.gradle |grep -Po \"${id}[^\\d]*\\K[^']*\"", returnStdout: true).trim()
+  env.BRANCH_REF="${branch}"
```

Comment:
if there a way to avoid such thing?

Reply:
there's only a couple of ways to get values into env vars. you could use the environment {} block and call a function to set its values

Reply:
no i mean, is there a way to compile string interpolaton with non string interpolation

Reply:
i just nedd to create the ENV crap so that it gets injected into the single quoted strings